pkg/generate/code/set_sdk.go

```diff
@@ -1636,15 +1636,27 @@ func setSDKAdaptiveResourceCollection(
 			out += fmt.Sprintf("%s\t%s.%s = aws.ToInt64Slice(%s)\n", indent, targetVarName, memberName, sourceAdaptedVarName)
 
 		}
-	} else if shape.Type == "map" &&
-		shape.KeyRef.Shape.Type == "string" &&
-		shape.ValueRef.Shape.Type == "string" {
-		out += fmt.Sprintf("%s\t%s.%s = aws.ToStringMap(%s)\n", indent, targetVarName, memberName, sourceAdaptedVarName)
+		} else if shape.Type == "map" &&
+		shape.KeyRef.Shape.Type == "string" {
+		mapType := getAWSMapType(shape.ValueRef.Shape.Type)
+		out += fmt.Sprintf("%s\t%s.%s = aws.To%sMap(%s)\n", indent, targetVarName, memberName, mapType, sourceAdaptedVarName)
 	}
-
 	return out
 }
 
+func getAWSMapType(valueType string) string {
```

Comment:
I'm thinking if we can use a more descriptive name here `resolveAWSMapValueType` or `nativeTypeToAWSType`, maybe a comment to the function can help as well